diff --git a/tools/ResearchProjectToolkit/javaparser/javaparser-core/src/main/javacc/java.jj b/tools/ResearchProjectToolkit/javaparser/javaparser-core/src/main/javacc/java.jj
index cbde7422b..c22487512 100644
--- a/tools/ResearchProjectToolkit/javaparser/javaparser-core/src/main/javacc/java.jj
+++ b/tools/ResearchProjectToolkit/javaparser/javaparser-core/src/main/javacc/java.jj
@@ -2984,7 +2984,7 @@ String Identifier():
         // Make sure the module info keywords don't interfere with normal Java parsing by matching them as normal identifiers.
         <MODULE> | <REQUIRES> | <TO> | <WITH> | <OPEN> | <OPENS> | <USES> | <EXPORTS> | <PROVIDES> | <TRANSITIVE> |
         // Make sure older Java versions parse
-        <ENUM> | <STRICTFP> | <YIELD> | <RECORD> |
+        <ENUM> | <STRICTFP> | <YIELD> | <RECORD> | <PERMITS> | <SEALED> |
         // An actual plain old identifier
         <IDENTIFIER>
     ) { ret = token.image; setTokenKind(IDENTIFIER);}
